src/Library/Plugin/Notice.php

```diff
@@ -0,0 +1,295 @@
+	 * @since SINCEVERSION
+	 *
+	 */
+	private function setId( $id ) {
+		$this->id = $id;
+	}
+
+	/**
+	 * Get slug of the page this feature is on.
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getPageSlug() {
+        return $this->pageSlug;
+    }
+
+	/**
+	 * Set pageSlug
+	 *
+	 * @param string $pageSlug Notice Page.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setPageSlug( string $pageSlug ) {
+		$this->pageSlug = $pageSlug;
+	}
+
+	/**
+	 * Get version of plugin this notice was added on
+	 *
+	 * @return string
+	 * @since SINCEVERSION
+	 */
+    public function getVersion() {
+        return $this->version;
+    }
+
+	/**
+	 * Set version of plugin this notice was added on
+	 *
+	 * @param string $version Version of plugin this notice was added on.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setVersion( string $version ) {
+		$this->version = $version;
+	}
+
+    /**
+     * Maybe Show
+     * 
+     * Returns true if this plugin's version number is
+     * greater than the first installed version, and feature's version number is 
+     * greater than or equal to current version
+     * @return bool
+     * @since SINCEVERSION
+     */
+    
+    public function maybeShow() {
+        $pluginVersion = $this->plugin->getPluginData()['Version'];
+        $versionIsNotFirst = $this->plugin->firstVersionCompare( $pluginVersion, '<' );
+        $featureIsNewer = version_compare( $this->version, $pluginVersion , '>=' );
+        return ( $featureIsNewer && $versionIsNotFirst);
+    }
+
+	/**
+	 * Get isUnread Value.
+	 *
+	 * @return bool
+	 * @since SINCEVERSION
+	 */
+    public function getIsUnread() {
+        if ( $this->maybeShow() ) {
+            return $this->isUnread;
+        }
+        return false;
+    }
+
+	/**
+	 * Set the notice as read or unread
+	 *
+	 * @param bool $isUnread Specifies if the Notice is Unread or not.
+	 * @since SINCEVERSION
+	 *
+	 */
+	public function setIsUnread( $isUnread ) {
+        $this->isUnread = $isUnread;
+        $this->updateNoticeOption();
+    }
+    
+    /**
+     * Determine if Notice Exists in Options Table
+     * 
+     * If the notice exists, then the return value is the index
+     * of that notice in the $option array. If the notice does not exist
+     * this returns false.
+     * 
+     * @return mixed 
+     * @since SINCEVERSION
+     */
+    public function alreadyExists( $noticeId ) {
+        $option = get_option( 'boldgrid_plugin_page_notices');
+        if ( $option ) {
+            foreach ( $option as $notice ) {
+                if ( $notice->getId() == $noticeId ) {
+                    return array_search($notice, $option);
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get Notice instance from wp_options row.
+     * 
+     * @param string $noticeId
+     * @return Notice
+     * @since SINCEVERSION
+     */
+    private function getNoticeFromOptions( $noticeId ) {
```

Comment:
Generally we don't want more than 3 words in a function name - this one is 4 - `getNoticeFromOptions`.

The code in this method seems very close to the code in `alreadyExists`. If we can, let's try not to duplicate the code.

I mentioned this above, but maybe we can have a helper class with a method of `$noticeHelper->getFromOption( $id )` or something similar.